src/Management/src/Endpoint/Middleware/EndpointMiddleware.cs

```diff
@@ -2,63 +2,91 @@
 // The .NET Foundation licenses this file to you under the Apache 2.0 License.
 // See the LICENSE file in the project root for more information.
 
+using System.Net;
 using System.Text.Json;
 using System.Text.Json.Serialization;
 using Microsoft.AspNetCore.Http;
 using Microsoft.Extensions.Logging;
 using Microsoft.Extensions.Options;
 using Steeltoe.Common;
+using Steeltoe.Management.Endpoint.CloudFoundry;
+using Steeltoe.Management.Endpoint.ContentNegotiation;
 using Steeltoe.Management.Endpoint.Health;
 using Steeltoe.Management.Endpoint.Metrics;
 using Steeltoe.Management.Endpoint.Options;
 
 namespace Steeltoe.Management.Endpoint.Middleware;
 
-public abstract class EndpointMiddleware<TResult> : IEndpointMiddleware
+public interface IEndpointMiddleware : IMiddleware
 {
-    protected ILogger Logger { get; }
-    protected IOptionsMonitor<ManagementEndpointOptions> ManagementOptions { get; }
+    HttpMiddlewareOptions EndpointOptions { get; }
 
-    public IEndpoint<TResult> Endpoint { get; set; }
+    bool ShouldInvoke(HttpContext context);
+}
 
-    public virtual IEndpointOptions EndpointOptions => Endpoint.Options;
+public abstract class EndpointMiddleware<TArgument, TResult> : IEndpointMiddleware
+{
+    protected ILogger Logger { get; }
+    protected IOptionsMonitor<ManagementEndpointOptions> ManagementEndpointOptions { get; }
+    public IEndpointHandler<TArgument, TResult> EndpointHandler { get; }
 
-    protected EndpointMiddleware(IOptionsMonitor<ManagementEndpointOptions> managementOptions, ILogger logger)
+    public HttpMiddlewareOptions EndpointOptions { get; }
```

Comment:
I tried but this is not possible because the IEndpointMiddleware does not have (and cannot have the EndpointHandler<TRequest, TResult> and we need the options at in the ActuatorEndpointMapper for each IEndpointMiddleware.